### PR TITLE
General platform-logic refactoring & Haiku support.

### LINKED ISSUE #1820: Haiku build support
I recently tried to build lmms on Haiku os, I ran into numerous cmake issues with finding the installed librarys for QT. Haiku has a customer package manager which may be cuasing some issues. 

any help appreciated, in fact a functioning daw may get some haiku devs to kick back code to the lmms project they are fairly good about upstreaming patches etc.

### PR DESCRIPTION
**Pull request by waddlesplash**
Fixes #1820.

### CODE REVIEW

**Comment by tresf** [comment]
This looks pretty good, but something has broken our win32 builds per:

**Comment by waddlesplash** [comment]
There, that should fix shared memory on Windows.

**Comment by tresf** [comment]
:+1: 

Since this adds a new `LMMS_BUILD_HAIKU`, I'd like @Lukas-W's input before merging.  Tremendous.  Thanks!

**Comment by diizy** [comment]
In lb302, it'd be better to just remove that MIN macro completely and replace it with qMin() to be consistent with rest of our codebase.

**Comment by waddlesplash** [comment]
Done.

File: include/lmms_math.h [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #include <cmath>
 using namespace std;
 
-#if defined (LMMS_BUILD_WIN32) || defined (LMMS_BUILD_APPLE) 
+#if defined (LMMS_BUILD_WIN32) || defined (LMMS_BUILD_APPLE) || defined(LMMS_BUILD_HAIKU)
```
Comment (Lukas-W) [inline include/lmms_math.h]: As I wrote #1822, I think this `#if` condition can just be removed and its body enabled on all platforms.

Comment (waddlesplash) [inline include/lmms_math.h]: Except some OSes may not have these as #defines, so you'll be overriding the OS version.

Comment (Lukas-W) [inline include/lmms_math.h]: That is a very valid argument.

**Comment by tresf** [comment]
@waddlesplash thanks as this cleans things up quite a bit.  We'll tag this commit if we have any fallout from a VST perspective.  Merging.

**Pull request merged**